From b46b7135e525bdf2f0e7cb641bc75c2a931cb11d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Piotr=20Kr=C3=B3l?= <piotr.krol@3mdeb.com>
Date: Wed, 8 May 2024 13:40:23 +0200
Subject: [PATCH 02/80] payloads/seabios: use PC Engines repository
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Piotr Król <piotr.krol@3mdeb.com>
diff --git a/configs/config.pcengines_apu2 b/configs/config.pcengines_apu2
index 5157ec1194e5..f8e362fa5e34 100644
--- a/configs/config.pcengines_apu2
+++ b/configs/config.pcengines_apu2
@@ -17,6 +17,8 @@ CONFIG_POST_IO_PORT=0x80
 CONFIG_PXE=y
 CONFIG_PXE_ROM_ID="8086,157b"
 CONFIG_SEABIOS_DEBUG_LEVEL=0
+CONFIG_SEABIOS_REVISION=y
+CONFIG_SEABIOS_REVISION_ID="rel-1.16.0.1"
 CONFIG_SEABIOS_SERCON_PORT_ADDR=0x3f8
 CONFIG_SUBSYSTEM_DEVICE_ID=0x0000
 CONFIG_SUBSYSTEM_VENDOR_ID=0x0000
diff --git a/configs/config.pcengines_apu3 b/configs/config.pcengines_apu3
index 89e4d995bd71..0604bb39f83a 100644
--- a/configs/config.pcengines_apu3
+++ b/configs/config.pcengines_apu3
@@ -10,4 +10,6 @@ CONFIG_NO_GFX_INIT=y
 CONFIG_PXE=y
 CONFIG_PXE_ROM_ID="8086,1539"
 CONFIG_SEABIOS_ADD_SERCON_PORT_FILE=y
+CONFIG_SEABIOS_REVISION=y
+CONFIG_SEABIOS_REVISION_ID="rel-1.16.0.1"
 CONFIG_VENDOR_PCENGINES=y
diff --git a/configs/config.pcengines_apu4 b/configs/config.pcengines_apu4
index ae119658be3e..7548f07e5573 100644
--- a/configs/config.pcengines_apu4
+++ b/configs/config.pcengines_apu4
@@ -18,6 +18,8 @@ CONFIG_POST_IO_PORT=0x80
 CONFIG_PXE=y
 CONFIG_PXE_ROM_ID="8086,1539"
 CONFIG_SEABIOS_DEBUG_LEVEL=0
+CONFIG_SEABIOS_REVISION=y
+CONFIG_SEABIOS_REVISION_ID="rel-1.16.0.1"
 CONFIG_SEABIOS_SERCON_PORT_ADDR=0x3f8
 CONFIG_SUBSYSTEM_DEVICE_ID=0x0000
 CONFIG_SUBSYSTEM_VENDOR_ID=0x0000
diff --git a/configs/config.pcengines_apu5 b/configs/config.pcengines_apu5
index 7e7f6558f643..3ad06d2cfbd5 100644
--- a/configs/config.pcengines_apu5
+++ b/configs/config.pcengines_apu5
@@ -10,4 +10,6 @@ CONFIG_NO_GFX_INIT=y
 CONFIG_PXE=y
 CONFIG_PXE_ROM_ID="8086,1539"
 CONFIG_SEABIOS_ADD_SERCON_PORT_FILE=y
+CONFIG_SEABIOS_REVISION=y
+CONFIG_SEABIOS_REVISION_ID="rel-1.16.0.1"
 CONFIG_VENDOR_PCENGINES=y
diff --git a/payloads/external/SeaBIOS/Makefile b/payloads/external/SeaBIOS/Makefile
index 7926f3fb90ad..5bdd526296ae 100644
--- a/payloads/external/SeaBIOS/Makefile
+++ b/payloads/external/SeaBIOS/Makefile
@@ -2,7 +2,7 @@ TAG-$(CONFIG_SEABIOS_MASTER)=origin/master
 TAG-$(CONFIG_SEABIOS_STABLE)=a6ed6b701f0a57db0569ab98b0661c12a6ec3ff8
 TAG-$(CONFIG_SEABIOS_REVISION)=$(CONFIG_SEABIOS_REVISION_ID)
 
-project_git_repo=https://review.coreboot.org/seabios.git
+project_git_repo=https://github.com/pcengines/seabios.git
 project_dir=seabios
 
 unexport KCONFIG_AUTOHEADER
