From 098596d69bb4411bc0e3fc559f50dd5f59555889 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Piotr=20Kr=C3=B3l?= <piotr.krol@3mdeb.com>
Date: Mon, 6 May 2024 17:29:04 +0200
Subject: [PATCH 14/80] mainboard/pcengines: Enable GENINTx as GPIO
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Piotr Król <piotr.krol@3mdeb.com>
---
 src/mainboard/pcengines/apu2/mainboard.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/mainboard/pcengines/apu2/mainboard.c b/src/mainboard/pcengines/apu2/mainboard.c
index 725375de1be0..c18d5c0f5fe1 100644
--- a/src/mainboard/pcengines/apu2/mainboard.c
+++ b/src/mainboard/pcengines/apu2/mainboard.c
@@ -38,6 +38,7 @@
 #define PM_RTC_CONTROL		0x56
 #define PM_RTC_SHADOW		0x5B
 #define PM_S_STATE_CONTROL	0xBA
+#define PM_PCI_CONFIG		0xEA
 
 #define SEC_REG_SERIAL_ADDR	0x1000
 #define MAX_SERIAL_LEN		10
@@ -287,8 +288,8 @@ static void mainboard_enable(struct device *dev)
 	/* Enable power on after power fail */
 	pm_write8(PM_RTC_SHADOW, pm_read8(PM_RTC_SHADOW) | (1 << 0));
 
-
-
+	/* Enable GENINTx as GPIO */
+	pm_write8(PM_PCI_CONFIG, 1);
 
 	/* Initialize the PIRQ data structures for consumption */
 	pirq_setup();
-- 
2.39.2

