From 2b744103e99f558a9c73dd76553d52b5feb8bdff Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Piotr=20Kr=C3=B3l?= <piotr.krol@3mdeb.com>
Date: Mon, 13 May 2024 16:07:36 +0200
Subject: [PATCH 21/80] mainboard/pcengines: use check_ehci0 in BiosCallOut
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Piotr Król <piotr.krol@3mdeb.com>
---
 src/mainboard/pcengines/apu2/BiosCallOuts.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/mainboard/pcengines/apu2/BiosCallOuts.c b/src/mainboard/pcengines/apu2/BiosCallOuts.c
index efd88d8c127c..5962392dc3eb 100644
--- a/src/mainboard/pcengines/apu2/BiosCallOuts.c
+++ b/src/mainboard/pcengines/apu2/BiosCallOuts.c
@@ -10,6 +10,7 @@
 #include "gpio_ftns.h"
 #include "imc.h"
 #include "hudson.h"
+#include "bios_knobs.h"
 
 static AGESA_STATUS board_ReadSpd_from_cbfs(UINT32 Func, UINTN Data, VOID *ConfigPtr);
 
@@ -74,7 +75,7 @@ void board_FCH_InitEnv(struct sysinfo *cb_NA, FCH_DATA_BLOCK *FchParams)
 		FchParams->Usb.Ehci1Enable = FALSE;
 	} else {
 		// Enable EHCI 0 (port 0 to 3)
-		FchParams->Usb.Ehci1Enable = TRUE;
+		FchParams->Usb.Ehci1Enable = check_ehci0();
 	}
 
 	// Enable EHCI 1 (port 4 to 7)
-- 
2.39.2

