From f1716651577218da0d6bf3de7e8d6c34c79ec3a9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Piotr=20Kr=C3=B3l?= <piotr.krol@3mdeb.com>
Date: Mon, 6 May 2024 11:55:33 +0200
Subject: [PATCH 22/80] mainboard/pcengines: use bios knobs
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Piotr Król <piotr.krol@3mdeb.com>
---
 src/mainboard/pcengines/apu2/mainboard.c | 57 ++++++++++++++++++------
 1 file changed, 43 insertions(+), 14 deletions(-)

diff --git a/src/mainboard/pcengines/apu2/mainboard.c b/src/mainboard/pcengines/apu2/mainboard.c
index ed5c2b2f90c9..066e2ac98c33 100644
--- a/src/mainboard/pcengines/apu2/mainboard.c
+++ b/src/mainboard/pcengines/apu2/mainboard.c
@@ -149,19 +149,49 @@ static void config_gpio_mux(void)
 {
 	struct device *uart, *gpio;
 
-	uart = dev_find_slot_pnp(SIO_PORT, NCT5104D_SP3);
-	gpio = dev_find_slot_pnp(SIO_PORT, NCT5104D_GPIO0);
-	if (uart)
-		uart->enabled = CONFIG(APU2_PINMUX_UART_C);
-	if (gpio)
-		gpio->enabled = CONFIG(APU2_PINMUX_GPIO0);
-
-	uart = dev_find_slot_pnp(SIO_PORT, NCT5104D_SP4);
-	gpio = dev_find_slot_pnp(SIO_PORT, NCT5104D_GPIO1);
-	if (uart)
-		uart->enabled = CONFIG(APU2_PINMUX_UART_D);
-	if (gpio)
-		gpio->enabled = CONFIG(APU2_PINMUX_GPIO1);
+	if (check_uartc()) {
+		printk(BIOS_INFO, "UARTC enabled\n");
+
+		uart = dev_find_slot_pnp(SIO_PORT, NCT5104D_SP3);
+		if (uart)
+			uart->enabled = 1;
+
+		gpio = dev_find_slot_pnp(SIO_PORT, NCT5104D_GPIO0);
+		if (gpio)
+			gpio->enabled = 0;
+	} else {
+		printk(BIOS_INFO, "UARTC disabled\n");
+
+		uart = dev_find_slot_pnp(SIO_PORT, NCT5104D_SP3);
+		if (uart)
+			uart->enabled = 0;
+
+		gpio = dev_find_slot_pnp(SIO_PORT, NCT5104D_GPIO0);
+		if (gpio)
+			gpio->enabled = 1;
+	}
+
+	if (check_uartd()) {
+		printk(BIOS_INFO, "UARTD enabled\n");
+
+		uart = dev_find_slot_pnp(SIO_PORT, NCT5104D_SP4);
+		if (uart)
+			uart->enabled = 1;
+
+		gpio = dev_find_slot_pnp(SIO_PORT, NCT5104D_GPIO1);
+		if (gpio)
+			gpio->enabled = 0;
+	} else {
+		printk(BIOS_INFO, "UARTD disabled\n");
+
+		uart = dev_find_slot_pnp(SIO_PORT, NCT5104D_SP4);
+		if (uart)
+			uart->enabled = 0;
+
+		gpio = dev_find_slot_pnp(SIO_PORT, NCT5104D_GPIO1);
+		if (gpio)
+			gpio->enabled = 1;
+	}
 }
 
 /**********************************************
@@ -259,7 +289,6 @@ static void mainboard_enable(struct device *dev)
 	if(scon) {
 		printk(BIOS_ALERT, "%d MB", total_mem);
 	}
-
 	//
 	// Read memory configuration from GPIO 49 and 50
 	//
-- 
2.39.2

