From a8d693c7d10ae3f67f224ff821f0dbaf26494294 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20=C5=BBygowski?= <michal.zygowski@3mdeb.com>
Date: Fri, 31 Aug 2018 10:37:01 +0200
Subject: [PATCH 30/80] payloads/external/iPXE/menu.ipxe: enable autoboot for
 all interfaces
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>
diff --git a/configs/config.pcengines_apu2 b/configs/config.pcengines_apu2
index 4cf8467b3997..32415365709b 100644
--- a/configs/config.pcengines_apu2
+++ b/configs/config.pcengines_apu2
@@ -16,7 +16,9 @@ CONFIG_NO_GFX_INIT=y
 CONFIG_PAYLOAD_CONFIGFILE="$(top)/src/mainboard/$(MAINBOARDDIR)/seabios_config"
 CONFIG_POST_IO_PORT=0x80
 CONFIG_PXE=y
+CONFIG_PXE_ADD_SCRIPT=y
 CONFIG_PXE_ROM_ID="8086,157b"
+CONFIG_PXE_SCRIPT="payloads/external/iPXE/menu.ipxe"
 CONFIG_SEABIOS_BOOTORDER_FILE="$(top)/src/mainboard/$(MAINBOARDDIR)/variants/$(CONFIG_VARIANT_DIR)/bootorder"
 CONFIG_SEABIOS_DEBUG_LEVEL=0
 CONFIG_SEABIOS_REVISION=y
diff --git a/configs/config.pcengines_apu3 b/configs/config.pcengines_apu3
index 91e3cb3d10e4..aae30959c114 100644
--- a/configs/config.pcengines_apu3
+++ b/configs/config.pcengines_apu3
@@ -17,7 +17,9 @@ CONFIG_NO_GFX_INIT=y
 CONFIG_PAYLOAD_CONFIGFILE="$(top)/src/mainboard/$(MAINBOARDDIR)/seabios_config"
 CONFIG_POST_IO_PORT=0x80
 CONFIG_PXE=y
+CONFIG_PXE_ADD_SCRIPT=y
 CONFIG_PXE_ROM_ID="8086,1539"
+CONFIG_PXE_SCRIPT="payloads/external/iPXE/menu.ipxe"
 CONFIG_SEABIOS_BOOTORDER_FILE="$(top)/src/mainboard/$(MAINBOARDDIR)/variants/$(CONFIG_VARIANT_DIR)/bootorder"
 CONFIG_SEABIOS_DEBUG_LEVEL=0
 CONFIG_SEABIOS_REVISION=y
diff --git a/configs/config.pcengines_apu4 b/configs/config.pcengines_apu4
index a393e27a8f7b..223e0e0a14aa 100644
--- a/configs/config.pcengines_apu4
+++ b/configs/config.pcengines_apu4
@@ -17,7 +17,9 @@ CONFIG_NO_GFX_INIT=y
 CONFIG_PAYLOAD_CONFIGFILE="$(top)/src/mainboard/$(MAINBOARDDIR)/seabios_config"
 CONFIG_POST_IO_PORT=0x80
 CONFIG_PXE=y
+CONFIG_PXE_ADD_SCRIPT=y
 CONFIG_PXE_ROM_ID="8086,1539"
+CONFIG_PXE_SCRIPT="payloads/external/iPXE/menu.ipxe"
 CONFIG_SEABIOS_BOOTORDER_FILE="$(top)/src/mainboard/$(MAINBOARDDIR)/variants/$(CONFIG_VARIANT_DIR)/bootorder"
 CONFIG_SEABIOS_DEBUG_LEVEL=0
 CONFIG_SEABIOS_REVISION=y
diff --git a/configs/config.pcengines_apu5 b/configs/config.pcengines_apu5
index 6169e26cbbbd..25723e63cdbb 100644
--- a/configs/config.pcengines_apu5
+++ b/configs/config.pcengines_apu5
@@ -17,7 +17,9 @@ CONFIG_NO_GFX_INIT=y
 CONFIG_PAYLOAD_CONFIGFILE="$(top)/src/mainboard/$(MAINBOARDDIR)/seabios_config"
 CONFIG_POST_IO_PORT=0x80
 CONFIG_PXE=y
+CONFIG_PXE_ADD_SCRIPT=y
 CONFIG_PXE_ROM_ID="8086,1539"
+CONFIG_PXE_SCRIPT="payloads/external/iPXE/menu.ipxe"
 CONFIG_SEABIOS_BOOTORDER_FILE="$(top)/src/mainboard/$(MAINBOARDDIR)/variants/$(CONFIG_VARIANT_DIR)/bootorder"
 CONFIG_SEABIOS_DEBUG_LEVEL=0
 CONFIG_SEABIOS_REVISION=y
diff --git a/payloads/external/iPXE/menu.ipxe b/payloads/external/iPXE/menu.ipxe
new file mode 100644
index 000000000000..1b5719588273
--- /dev/null
+++ b/payloads/external/iPXE/menu.ipxe
@@ -0,0 +1,17 @@
+#!ipxe
+:MENU
+menu
+item --gap -- ---------------- iPXE boot menu ----------------
+item shell          ipxe shell
+item boot           autoboot
+choose --default boot --timeout 3000 target && goto ${target}
+
+:boot
+autoboot
+goto MENU
+
+:shell
+shell ||
+goto MENU
+
+autoboot
