From a8d693c7d10ae3f67f224ff821f0dbaf26494294 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20=C5=BBygowski?= <michal.zygowski@3mdeb.com>
Date: Fri, 31 Aug 2018 10:37:01 +0200
Subject: [PATCH 30/80] payloads/external/iPXE/menu.ipxe: enable autoboot for
 all interfaces
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>
diff --git a/configs/config.pcengines_seabios_apu2 b/configs/config.pcengines_seabios_apu2
index 5c091dd76d96..b17af2b2c5a0 100644
--- a/configs/config.pcengines_seabios_apu2
+++ b/configs/config.pcengines_seabios_apu2
@@ -9,6 +9,7 @@ CONFIG_DEFAULT_CONSOLE_LOGLEVEL_1=y
 CONFIG_HUDSON_SATA_MODE=2
 CONFIG_I2C_TRANSFER_TIMEOUT_US=500000
 CONFIG_IPXE_ADD_SCRIPT=y
+CONFIG_IPXE_SCRIPT="payloads/external/iPXE/menu.ipxe"
 CONFIG_LOCALVERSION="v24.05.00.01"
 CONFIG_MEMTEST86PLUS_V5=y
 CONFIG_MEMTEST_SECONDARY_PAYLOAD=y
diff --git a/configs/config.pcengines_seabios_apu3 b/configs/config.pcengines_seabios_apu3
index ae371050a17e..d021c93fa08c 100644
--- a/configs/config.pcengines_seabios_apu3
+++ b/configs/config.pcengines_seabios_apu3
@@ -10,6 +10,7 @@ CONFIG_DEFAULT_CONSOLE_LOGLEVEL_1=y
 CONFIG_HUDSON_SATA_MODE=2
 CONFIG_I2C_TRANSFER_TIMEOUT_US=500000
 CONFIG_IPXE_ADD_SCRIPT=y
+CONFIG_IPXE_SCRIPT="payloads/external/iPXE/menu.ipxe"
 CONFIG_LOCALVERSION="v24.05.00.01"
 CONFIG_MEMTEST86PLUS_V5=y
 CONFIG_MEMTEST_SECONDARY_PAYLOAD=y
diff --git a/configs/config.pcengines_seabios_apu4 b/configs/config.pcengines_seabios_apu4
index 7db3f16b9af2..5bdf4490a909 100644
--- a/configs/config.pcengines_seabios_apu4
+++ b/configs/config.pcengines_seabios_apu4
@@ -10,6 +10,7 @@ CONFIG_DEFAULT_CONSOLE_LOGLEVEL_1=y
 CONFIG_HUDSON_SATA_MODE=2
 CONFIG_I2C_TRANSFER_TIMEOUT_US=500000
 CONFIG_IPXE_ADD_SCRIPT=y
+CONFIG_IPXE_SCRIPT="payloads/external/iPXE/menu.ipxe"
 CONFIG_LOCALVERSION="v24.05.00.01"
 CONFIG_MEMTEST86PLUS_V5=y
 CONFIG_MEMTEST_SECONDARY_PAYLOAD=y
diff --git a/payloads/external/iPXE/menu.ipxe b/payloads/external/iPXE/menu.ipxe
new file mode 100644
index 000000000000..1b5719588273
--- /dev/null
+++ b/payloads/external/iPXE/menu.ipxe
@@ -0,0 +1,17 @@
+#!ipxe
+:MENU
+menu
+item --gap -- ---------------- iPXE boot menu ----------------
+item shell          ipxe shell
+item boot           autoboot
+choose --default boot --timeout 3000 target && goto ${target}
+
+:boot
+autoboot
+goto MENU
+
+:shell
+shell ||
+goto MENU
+
+autoboot
