From 7782d1b612aa674ccf67f23d8ecd126662fd4337 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20=C5=BBygowski?= <michal.zygowski@3mdeb.com>
Date: Thu, 22 Nov 2018 12:29:52 +0100
Subject: [PATCH 33/80] Makefile.inc: include microcode patch inside firmware
 image
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>
---
 Makefile.inc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Makefile.inc b/Makefile.inc
index 2b5d742a2cb8..039db5b91ef0 100644
--- a/Makefile.inc
+++ b/Makefile.inc
@@ -1151,6 +1151,9 @@ ifeq ($(CONFIG_INTEL_ADD_TOP_SWAP_BOOTBLOCK),y)
 	$(IFITTOOL) -f $@.tmp -D $(TS_OPTIONS) -r COREBOOT
 endif # CONFIG_INTEL_ADD_TOP_SWAP_BOOTBLOCK
 endif # CONFIG_CPU_INTEL_FIRMWARE_INTERFACE_TABLE
+ifeq ($(CONFIG_AGESA_UCODE_EXPERIMENTAL),y)
+	dd if=$(CONFIG_CPU_UCODE_BINARIES) of=$@.tmp bs=1 seek=6665692 count=3424 conv=notrunc 2> /dev/null
+endif
 	mv $@.tmp $@
 	@printf "    CBFSLAYOUT  $(subst $(obj)/,,$(@))\n\n"
 	$(CBFSTOOL) $@ layout
-- 
2.39.2

