From 64a4ffe3f6fcaf8b3bac59c7d44dbca56e9c8b20 Mon Sep 17 00:00:00 2001
From: Krystian Hebel <krystian.hebel@3mdeb.com>
Date: Fri, 28 Dec 2018 16:54:40 +0100
Subject: [PATCH 34/80] payloads/iPXE: option for reproducible building

Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>
diff --git a/configs/config.pcengines_seabios_apu2 b/configs/config.pcengines_seabios_apu2
index b17af2b2c5a0..8d9e186623f8 100644
--- a/configs/config.pcengines_seabios_apu2
+++ b/configs/config.pcengines_seabios_apu2
@@ -9,6 +9,7 @@ CONFIG_DEFAULT_CONSOLE_LOGLEVEL_1=y
 CONFIG_HUDSON_SATA_MODE=2
 CONFIG_I2C_TRANSFER_TIMEOUT_US=500000
 CONFIG_IPXE_ADD_SCRIPT=y
+CONFIG_IPXE_CUSTOM_BUILD_ID="12345678"
 CONFIG_IPXE_SCRIPT="payloads/external/iPXE/menu.ipxe"
 CONFIG_LOCALVERSION="v24.05.00.01"
 CONFIG_MEMTEST86PLUS_V5=y
diff --git a/configs/config.pcengines_seabios_apu3 b/configs/config.pcengines_seabios_apu3
index d021c93fa08c..3f3ca9b0aa57 100644
--- a/configs/config.pcengines_seabios_apu3
+++ b/configs/config.pcengines_seabios_apu3
@@ -10,6 +10,7 @@ CONFIG_DEFAULT_CONSOLE_LOGLEVEL_1=y
 CONFIG_HUDSON_SATA_MODE=2
 CONFIG_I2C_TRANSFER_TIMEOUT_US=500000
 CONFIG_IPXE_ADD_SCRIPT=y
+CONFIG_IPXE_CUSTOM_BUILD_ID="12345678"
 CONFIG_IPXE_SCRIPT="payloads/external/iPXE/menu.ipxe"
 CONFIG_LOCALVERSION="v24.05.00.01"
 CONFIG_MEMTEST86PLUS_V5=y
diff --git a/configs/config.pcengines_seabios_apu4 b/configs/config.pcengines_seabios_apu4
index 5bdf4490a909..c1fcf2a157df 100644
--- a/configs/config.pcengines_seabios_apu4
+++ b/configs/config.pcengines_seabios_apu4
@@ -10,6 +10,7 @@ CONFIG_DEFAULT_CONSOLE_LOGLEVEL_1=y
 CONFIG_HUDSON_SATA_MODE=2
 CONFIG_I2C_TRANSFER_TIMEOUT_US=500000
 CONFIG_IPXE_ADD_SCRIPT=y
+CONFIG_IPXE_CUSTOM_BUILD_ID="12345678"
 CONFIG_IPXE_SCRIPT="payloads/external/iPXE/menu.ipxe"
 CONFIG_LOCALVERSION="v24.05.00.01"
 CONFIG_MEMTEST86PLUS_V5=y
diff --git a/payloads/external/Makefile.mk b/payloads/external/Makefile.mk
index d2961e0fec5b..eb6c014d4684 100644
--- a/payloads/external/Makefile.mk
+++ b/payloads/external/Makefile.mk
@@ -431,6 +431,7 @@ payloads/external/iPXE/ipxe/ipxe.rom ipxe: $(DOTCONFIG) $(IPXE_CONFIG_SCRIPT)
 	CONFIG_TTYS0_BAUD=$(CONFIG_TTYS0_BAUD) \
 	CONFIG_PXE_CUSTOM_GENERAL_H=$(CONFIG_PXE_CUSTOM_GENERAL_H) \
 	CONFIG_PXE_CUSTOM_BOOTMENU_FILE=$(CONFIG_PXE_CUSTOM_BOOTMENU_FILE) \
+	CONFIG_IPXE_CUSTOM_BUILD_ID=$(CONFIG_IPXE_CUSTOM_BUILD_ID) \
 	CONFIG_SCRIPT=$(IPXE_CONFIG_SCRIPT) \
 	CONFIG_HAS_SCRIPT=$(CONFIG_IPXE_ADD_SCRIPT) \
 	CONFIG_IPXE_NO_PROMPT=$(CONFIG_IPXE_NO_PROMPT) \
diff --git a/payloads/external/iPXE/Kconfig b/payloads/external/iPXE/Kconfig
index 3f0107374350..b8c30d6c0307 100644
--- a/payloads/external/iPXE/Kconfig
+++ b/payloads/external/iPXE/Kconfig
@@ -83,6 +83,13 @@ config IPXE_SERIAL_CONSOLE
 
 	  Unselect to let only SeaBIOS handle printing output.
 
+config IPXE_CUSTOM_BUILD_ID
+	string "iPXE custom build_id variable"
+	default ""
+	help
+	  This option allows user to customize build_id for reproducible builds.
+	  It is 32-bit hexadecimal number without "0x" prefix.
+
 config IPXE_NO_PROMPT
 	bool "Do not show prompt to boot from PXE"
 	default n
diff --git a/payloads/external/iPXE/Makefile b/payloads/external/iPXE/Makefile
index 83fa6b911b10..75aab39074b4 100644
--- a/payloads/external/iPXE/Makefile
+++ b/payloads/external/iPXE/Makefile
@@ -18,6 +18,10 @@ unexport KCONFIG_SPLITCONFIG
 unexport KCONFIG_TRISTATE
 unexport KCONFIG_NEGATIVES
 
+ifneq ($(CONFIG_IPXE_CUSTOM_BUILD_ID),)
+PXE_MAKE_OPTS := BUILD_ID_CMD="echo 0x$(CONFIG_IPXE_CUSTOM_BUILD_ID)"
+endif
+
 all: build
 
 $(project_dir):
@@ -74,7 +78,7 @@ endif
 build: config $(CONFIG_SCRIPT)
 ifeq ($(CONFIG_HAS_SCRIPT),y)
 	echo "    MAKE       $(project_name) $(TAG-y) EMBED=$(CONFIG_SCRIPT)"
-	$(MAKE) -C $(project_dir)/src bin/$(PXE_ROM_PCI_ID).rom EMBED=$(CONFIG_SCRIPT)
+	$(MAKE) -C $(project_dir)/src bin/$(PXE_ROM_PCI_ID).rom EMBED=$(CONFIG_SCRIPT) $(PXE_MAKE_OPTS)
 else
 	echo "    MAKE       $(project_name) $(TAG-y)"
 	$(MAKE) -C $(project_dir)/src bin/$(PXE_ROM_PCI_ID).rom $(PXE_MAKE_OPTS)
