From ade0a0cbd3df2bb91380fb014ec851b4e73414bb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20=C5=BBygowski?= <michal.zygowski@3mdeb.com>
Date: Fri, 31 Aug 2018 10:37:01 +0200
Subject: [PATCH 35/88] payloads/external/iPXE/menu.ipxe: enable autoboot for
 all interfaces
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Upstream-Status: Pending
Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>
diff --git a/configs/config.pcengines_seabios_apu2 b/configs/config.pcengines_seabios_apu2
index f85c85905afb..036ede098bf2 100644
--- a/configs/config.pcengines_seabios_apu2
+++ b/configs/config.pcengines_seabios_apu2
@@ -9,6 +9,7 @@ CONFIG_DEFAULT_CONSOLE_LOGLEVEL_1=y
 CONFIG_HUDSON_SATA_MODE=2
 CONFIG_I2C_TRANSFER_TIMEOUT_US=500000
 CONFIG_IPXE_ADD_SCRIPT=y
+CONFIG_IPXE_SCRIPT="payloads/external/iPXE/menu.ipxe"
 CONFIG_LOCALVERSION="v25.09.00.01"
 CONFIG_MEMTEST86PLUS_V5=y
 CONFIG_MEMTEST_SECONDARY_PAYLOAD=y
diff --git a/configs/config.pcengines_seabios_apu3 b/configs/config.pcengines_seabios_apu3
index 7b9ac18c90e3..99ed671ed5a9 100644
--- a/configs/config.pcengines_seabios_apu3
+++ b/configs/config.pcengines_seabios_apu3
@@ -10,6 +10,7 @@ CONFIG_DEFAULT_CONSOLE_LOGLEVEL_1=y
 CONFIG_HUDSON_SATA_MODE=2
 CONFIG_I2C_TRANSFER_TIMEOUT_US=500000
 CONFIG_IPXE_ADD_SCRIPT=y
+CONFIG_IPXE_SCRIPT="payloads/external/iPXE/menu.ipxe"
 CONFIG_LOCALVERSION="v25.09.00.01"
 CONFIG_MEMTEST86PLUS_V5=y
 CONFIG_MEMTEST_SECONDARY_PAYLOAD=y
diff --git a/configs/config.pcengines_seabios_apu4 b/configs/config.pcengines_seabios_apu4
index c9c296f3d151..d4aa57ac3639 100644
--- a/configs/config.pcengines_seabios_apu4
+++ b/configs/config.pcengines_seabios_apu4
@@ -10,6 +10,7 @@ CONFIG_DEFAULT_CONSOLE_LOGLEVEL_1=y
 CONFIG_HUDSON_SATA_MODE=2
 CONFIG_I2C_TRANSFER_TIMEOUT_US=500000
 CONFIG_IPXE_ADD_SCRIPT=y
+CONFIG_IPXE_SCRIPT="payloads/external/iPXE/menu.ipxe"
 CONFIG_LOCALVERSION="v25.09.00.01"
 CONFIG_MEMTEST86PLUS_V5=y
 CONFIG_MEMTEST_SECONDARY_PAYLOAD=y
