From 67e42e3dcd5669cfab53265004f19fd9ed7cf2ac Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Piotr=20Kr=C3=B3l?= <piotr.krol@3mdeb.com>
Date: Mon, 6 May 2024 13:26:50 +0200
Subject: [PATCH 38/85] mainboard/pcengines: add boost knob check
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Upstream-Status: Inappropriate [Should utilize existing option backend]
Signed-off-by: Piotr Król <piotr.krol@3mdeb.com>
---
 src/mainboard/pcengines/apu2/bios_knobs.c | 25 +++++++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/src/mainboard/pcengines/apu2/bios_knobs.c b/src/mainboard/pcengines/apu2/bios_knobs.c
index 1f237fc4d6..5ae0da8e3a 100644
--- a/src/mainboard/pcengines/apu2/bios_knobs.c
+++ b/src/mainboard/pcengines/apu2/bios_knobs.c
@@ -170,6 +170,31 @@ int check_com2(void)
 	return 0;
 }
 
+int check_boost(void)
+{
+	u8 boost;
+
+	//
+	// Find the CPU boost item
+	//
+	boost = check_knob_value("boosten");
+
+	switch (boost) {
+	case 0:
+		return 0;
+		break;
+	case 1:
+		return 1;
+		break;
+	default:
+		printk(BIOS_INFO,
+			"Missing or invalid boost knob, disable CPU boost.\n");
+		break;
+	}
+
+	return 0;
+}
+
 static u8 check_uart(char uart_letter)
 {
 	u8 uarten;
-- 
2.49.0

