From 7b833a1de756e84ecae0beeff07947ef099b9d14 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20=C5=BBygowski?= <michal.zygowski@3mdeb.com>
Date: Sun, 28 Jun 2020 19:21:45 +0200
Subject: [PATCH 65/88] coreboot_table: include DRTM TPM log in coreboot tables
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Upstream-Status: Submitted [CB:5155157981]
Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>
diff --git a/src/commonlib/include/commonlib/coreboot_tables.h b/src/commonlib/include/commonlib/coreboot_tables.h
index ab57ed0270b8..3991e2ca48e2 100644
--- a/src/commonlib/include/commonlib/coreboot_tables.h
+++ b/src/commonlib/include/commonlib/coreboot_tables.h
@@ -90,6 +90,7 @@ enum {
 	LB_TAG_EFI_FW_INFO		= 0x0045,
 	LB_TAG_CAPSULE			= 0x0046,
 	LB_TAG_CFR_ROOT			= 0x0047,
+	LB_TAG_DRTM_LOG			= 0x0048,
 	/* The following options are CMOS-related */
 	LB_TAG_CMOS_OPTION_TABLE	= 0x00c8,
 	LB_TAG_OPTION			= 0x00c9,
diff --git a/src/lib/coreboot_table.c b/src/lib/coreboot_table.c
index d8ecdce868e6..5878782099b6 100644
--- a/src/lib/coreboot_table.c
+++ b/src/lib/coreboot_table.c
@@ -271,6 +271,7 @@ static void add_cbmem_pointers(struct lb_header *header)
 		{CBMEM_ID_VPD, LB_TAG_VPD},
 		{CBMEM_ID_WIFI_CALIBRATION, LB_TAG_WIFI_CALIBRATION},
 		{CBMEM_ID_TPM_CB_LOG, LB_TAG_TPM_CB_LOG},
+		{CBMEM_ID_DRTM_LOG, LB_TAG_DRTM_LOG},
 		{CBMEM_ID_FMAP, LB_TAG_FMAP},
 		{CBMEM_ID_VBOOT_WORKBUF, LB_TAG_VBOOT_WORKBUF},
 		{CBMEM_ID_TYPE_C_INFO, LB_TAG_TYPE_C_INFO},
