From 19e672236e41c98101f9316ab4c10e26632d9cf7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20=C5=BBygowski?= <michal.zygowski@3mdeb.com>
Date: Sun, 28 Jun 2020 19:22:30 +0200
Subject: [PATCH 66/88] util/cbmem/cbmem.c: implement DRTM TPM log parser
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Upstream-Status: Submitted [CB:51583]
Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>
Signed-off-by: Krystian Hebel <krystian.hebel@3mdeb.com>
Signed-off-by: Michał Kopeć <michal.kopec@3mdeb.com>
Signed-off-by: Marek Kasiewicz <marek.kasiewicz@3mdeb.com>
diff --git a/util/cbmem/cbmem.c b/util/cbmem/cbmem.c
index 018ea871f9ca..4f9b085bb9ab 100644
--- a/util/cbmem/cbmem.c
+++ b/util/cbmem/cbmem.c
@@ -35,6 +35,11 @@
 /* Global verbosity level for debug() macro. */
 int cbmem_util_verbose;
 
+/* Globals specifying memory to search in devmem backed  */
+int address_specified = 0;
+unsigned long long possible_address = 0LL;
+size_t cb_table_size = 0;
+
 #if defined(linux) && (defined(__i386__) || defined(__x86_64__))
 /*
  * read CPU frequency from a sysfs file, return an frequency in Megahertz as
@@ -702,6 +707,17 @@ static void dump_tpm_log(void)
 		dump_tpm_cb_log();
 }
 
+static void dump_drtm_log(void)
+{
+	uint8_t *buf;
+
+	if (cbmem_drv_get_cbmem_entry(CBMEM_ID_DRTM_LOG, &buf, NULL, NULL)) {
+		printf("DRTM ");
+		dump_tpm_std_log(buf);
+		free(buf);
+	}
+}
+
 enum console_print_type {
 	CONSOLE_PRINT_FULL = 0,
 	CONSOLE_PRINT_LAST,
@@ -1054,8 +1070,10 @@ static void print_version(void)
 
 static void print_usage(const char *name, int exit_code)
 {
-	printf("usage: %s [-cCltTLxVvh?]\n", name);
+	printf("usage: %s [-A:s:c12B:CltTSa:LdxVvh?r:]\n", name);
 	printf("\n"
+	     "   -A | --addr address:              set base address\n"
+	     "   -s | --size size:                 set table size. Change is applied only if address is also specified\n"
 	     "   -c | --console:                   print cbmem console\n"
 	     "   -1 | --oneboot:                   print cbmem console for last boot only\n"
 	     "   -2 | --2ndtolast:                 print cbmem console for the boot that came before the last one only\n"
@@ -1070,6 +1088,7 @@ static void print_usage(const char *name, int exit_code)
 	     "   -S | --stacked-timestamps:        print stacked timestamps (e.g. for flame graph tools)\n"
 	     "   -a | --add-timestamp ID:          append timestamp with ID\n"
 	     "   -L | --tcpa-log                   print TPM log\n"
+	     "   -d | --drtm-log                   print DRTM TPM log\n"
 	     "   -V | --verbose:                   verbose (debugging) output\n"
 	     "   -v | --version:                   print the version\n"
 	     "   -h | --help:                      print this help\n"
@@ -1086,6 +1105,7 @@ int main(int argc, char** argv)
 	int print_hexdump = 0;
 	int print_rawdump = 0;
 	int print_tcpa_log = 0;
+	int print_drtm_log = 0;
 	enum timestamps_print_type timestamp_type = TIMESTAMPS_PRINT_NONE;
 	enum console_print_type console_type = CONSOLE_PRINT_FULL;
 	unsigned int rawdump_id = 0;
@@ -1096,6 +1116,8 @@ int main(int argc, char** argv)
 
 	int opt, option_index = 0;
 	static struct option long_options[] = {
+		{"addr", 0, 0, 'A'},
+		{"size", 0, 0, 's'},
 		{"console", 0, 0, 'c'},
 		{"backend", required_argument, 0, 'b'},
 		{"oneboot", 0, 0, '1'},
@@ -1104,6 +1126,7 @@ int main(int argc, char** argv)
 		{"coverage", 0, 0, 'C'},
 		{"list", 0, 0, 'l'},
 		{"tcpa-log", 0, 0, 'L'},
+		{"drtm-log", 0, 0, 'd'},
 		{"timestamps", 0, 0, 't'},
 		{"parseable-timestamps", 0, 0, 'T'},
 		{"stacked-timestamps", 0, 0, 'S'},
@@ -1115,9 +1138,19 @@ int main(int argc, char** argv)
 		{"help", 0, 0, 'h'},
 		{0, 0, 0, 0}
 	};
-	while ((opt = getopt_long(argc, argv, "cb:12B:CltTSa:LxVvh?r:",
+	while ((opt = getopt_long(argc, argv, "A:s:cb:12B:CltTSa:LdxVvh?r:",
 				  long_options, &option_index)) != EOF) {
 		switch (opt) {
+		case 'A':
+			if (optarg) {
+				possible_address = strtoull(optarg, NULL, 0);
+				address_specified = 1;
+			}
+			break;
+		case 's':
+			if (optarg)
+				cb_table_size = strtoul(optarg, NULL, 0);
+			break;
 		case 'c':
 			print_console = 1;
 			print_defaults = 0;
@@ -1158,6 +1191,10 @@ int main(int argc, char** argv)
 			print_tcpa_log = 1;
 			print_defaults = 0;
 			break;
+		case 'd':
+			print_drtm_log = 1;
+			print_defaults = 0;
+			break;
 		case 'x':
 			print_hexdump = 1;
 			print_defaults = 0;
@@ -1246,6 +1283,9 @@ int main(int argc, char** argv)
 	if (print_tcpa_log)
 		dump_tpm_log();
 
+	if (print_drtm_log)
+		dump_drtm_log();
+
 	cbmem_drv_terminate();
 
 	return 0;
diff --git a/util/cbmem/cbmem_util.h b/util/cbmem/cbmem_util.h
index 3df0eabe7d34..8673584cc2fb 100644
--- a/util/cbmem/cbmem_util.h
+++ b/util/cbmem/cbmem_util.h
@@ -8,6 +8,9 @@
 #include <commonlib/bsd/helpers.h>
 
 extern int cbmem_util_verbose;
+extern int address_specified;
+extern unsigned long long possible_address;
+extern size_t cb_table_size;
 
 #define debug(x...)                                                            \
 	do {                                                                   \
diff --git a/util/cbmem/devmem_drv.c b/util/cbmem/devmem_drv.c
index 249f44a4cd33..b3f32c1b2a64 100644
--- a/util/cbmem/devmem_drv.c
+++ b/util/cbmem/devmem_drv.c
@@ -414,11 +414,15 @@ bool cbmem_devmem_init(bool writeable)
 #else
 	unsigned long long possible_base_addresses[] = {0, 0xf0000};
 
-	/* Find and parse coreboot table */
-	for (size_t j = 0; j < ARRAY_SIZE(possible_base_addresses); j++) {
-		if (!parse_cbtable(possible_base_addresses[j], 0))
-			break;
-	}
+	if (address_specified) {
+		parse_cbtable(possible_address, cb_table_size);
+	} else {
+		for (size_t j = 0; j < ARRAY_SIZE(possible_base_addresses); j++) {
+			if (!parse_cbtable(possible_base_addresses[j], 0))
+				break;
+		}
+ 	}
+
 #endif
 
 	if (mapping_virt(&lbtable_mapping) == NULL) {
