From f935f4ea0669a2d4e746ae455f3ac8862bfef108 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20=C5=BBygowski?= <michal.zygowski@3mdeb.com>
Date: Thu, 29 May 2025 16:18:18 +0200
Subject: [PATCH 87/88] use 3rdparty/dasharo-blobs/dasharo/dasharo.ipxe


Upstream-Status: Inappropriate [Custom submodule]
Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

diff --git a/configs/config.pcengines_seabios_apu2 b/configs/config.pcengines_seabios_apu2
index 776a24011ca8..22f0f399deec 100644
--- a/configs/config.pcengines_seabios_apu2
+++ b/configs/config.pcengines_seabios_apu2
@@ -9,6 +9,7 @@ CONFIG_DEFAULT_CONSOLE_LOGLEVEL_1=y
 CONFIG_HUDSON_SATA_MODE=2
 CONFIG_I2C_TRANSFER_TIMEOUT_US=500000
 CONFIG_IPXE_ADD_SCRIPT=y
+CONFIG_IPXE_SCRIPT="3rdparty/dasharo-blobs/dasharo/dasharo.ipxe"
 CONFIG_LOCALVERSION="v25.09.00.01"
 CONFIG_MEMTEST86PLUS_V5=y
 CONFIG_MEMTEST_SECONDARY_PAYLOAD=y
diff --git a/configs/config.pcengines_seabios_apu3 b/configs/config.pcengines_seabios_apu3
index 290d27872723..a15c2bab503d 100644
--- a/configs/config.pcengines_seabios_apu3
+++ b/configs/config.pcengines_seabios_apu3
@@ -10,6 +10,7 @@ CONFIG_DEFAULT_CONSOLE_LOGLEVEL_1=y
 CONFIG_HUDSON_SATA_MODE=2
 CONFIG_I2C_TRANSFER_TIMEOUT_US=500000
 CONFIG_IPXE_ADD_SCRIPT=y
+CONFIG_IPXE_SCRIPT="3rdparty/dasharo-blobs/dasharo/dasharo.ipxe"
 CONFIG_LOCALVERSION="v25.09.00.01"
 CONFIG_MEMTEST86PLUS_V5=y
 CONFIG_MEMTEST_SECONDARY_PAYLOAD=y
diff --git a/configs/config.pcengines_seabios_apu4 b/configs/config.pcengines_seabios_apu4
index ca8d340f47c1..04188f82bb42 100644
--- a/configs/config.pcengines_seabios_apu4
+++ b/configs/config.pcengines_seabios_apu4
@@ -10,6 +10,7 @@ CONFIG_DEFAULT_CONSOLE_LOGLEVEL_1=y
 CONFIG_HUDSON_SATA_MODE=2
 CONFIG_I2C_TRANSFER_TIMEOUT_US=500000
 CONFIG_IPXE_ADD_SCRIPT=y
+CONFIG_IPXE_SCRIPT="3rdparty/dasharo-blobs/dasharo/dasharo.ipxe"
 CONFIG_LOCALVERSION="v25.09.00.01"
 CONFIG_MEMTEST86PLUS_V5=y
 CONFIG_MEMTEST_SECONDARY_PAYLOAD=y
